libcxx/test/std/time/time.zone/time.zone.zonedtime/time.zone.zonedtime.members/operator_local_time.pass.cpp

```diff
@@ -0,0 +1,135 @@
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  {
+    using duration         = cr::microseconds;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<duration>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  {
+    using duration         = cr::milliseconds;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<duration>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  // common_type_t<seconds, seconds> -> seconds
+  {
+    using duration         = cr::seconds;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<duration>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  // common_type_t<duration, seconds> -> seconds
+  {
+    using duration         = cr::days;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<cr::seconds>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  {
+    using duration         = cr::weeks;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<cr::seconds>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  {
+    using duration         = cr::months;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<cr::seconds>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
+
+    std::same_as<local_time_point> decltype(auto) time = static_cast<local_time_point>(zt);
+    assert(time == local_time_point{duration{42} - cr::hours{1}});
+  }
+  {
+    using duration         = cr::years;
+    using time_point       = cr::sys_time<duration>;
+    using local_time_point = cr::local_time<cr::seconds>;
+    cr::zoned_time<duration> zt{"Etc/GMT+1", time_point{duration{42}}};
```

Comment:
Note that you could make all of those `zt`'s `const` since the method is `const`.

Reply:
As mentioned during the review. We typically don't use `const` a lot in our code base so it feels odd to do it here.